### PR TITLE
Run cosign commands in parallel [CLOUDDST-23799]

### PR DESCRIPTION
**Pull request by zxiong**

### CODE REVIEW

File: src/pubtools/sign/signers/cosignsigner.py [hunk 1]
```diff
@@ -275,10 +274,16 @@ def container_sign(self: CosignSigner, operation: ContainerSignOperation) -> Sig
...
-        for args in ref_args:
-            outputs.append(run_command(common_args + args, env=env_vars, tries=self.retries))
+        run_command_args = [
+            FData(
+                args=[common_args + args],
+                kwargs={"env": env_vars, "tries": self.retries},
+            )
+            for args in ref_args
+        ]
+        ret = run_in_parallel(run_command, run_command_args)
```
Comment (emilyzheng) [inline src/pubtools/sign/signers/cosignsigner.py:284]: Running in parallel for the same reference may cause conflict (multiple requests writing to the same signature, not all are written successfully). Could you ensure requests of each reference are run in line?

Comment (zxiong) [reply src/pubtools/sign/signers/cosignsigner.py]: Fixing it by grouping cosign command by reference, then running groups simultaneously and executing sequentially in each group.

**Review by emilyzheng — APPROVED** [review]

**Comment by midnightercz** [comment]
Did you run into any struggles when running it in parallel? When I was testing signing in parallel quay started refusing my requests when I set parallelism level too high.

**Comment by zxiong** [comment]
> Did you run into any struggles when running it in parallel? When I was testing signing in parallel quay started refusing my requests when I set parallelism level too high.

Seem no. I did more tests with 20 and 30 threads. The pub tasks are closed without errors in cosign. You can check the pub task logs in the JIRA task comments.

**Review by midnightercz — APPROVED** [review]

**Pull request merged**